vach/src/writer/builder/config.rs

```diff
@@ -1,17 +1,48 @@
-use crate::global::{flags::Flags, result::InternalResult};
+use crate::global::{flags::Flags, result::InternalResult, reg_entry::RegistryEntry};
 use std::io;
 use ed25519_dalek as esdalek;
+use std::fmt::Debug;
+
+type OptionalCallback = Option<Box<dyn Fn(&String, usize, &RegistryEntry)>>;
 
 /// Allows for the customization of valid `vach` archives during their construction.
 /// Such as custom `MAGIC`, custom `Header` flags and signing by providing a keypair.
-#[derive(Debug)]
 pub struct BuilderConfig {
 	/// Used to write a unique magic sequence into the write target.
 	pub magic: [u8; crate::MAGIC_LENGTH],
 	/// Flags to be written into the `Header` section of the write target.
 	pub flags: Flags,
 	/// An optional keypair. If a key is provided, then the write target will have signatures for tamper verification.
 	pub keypair: Option<esdalek::Keypair>,
+	/// An optional callback that is called every time a `Leaf` finishes processing
+	/// The type signature is:
+	/// ```ignore
+	/// type OptionalCallback = Option<Box<dyn Fn(&String, usize, &RegistryEntry)>>;
```

Comment:
we should probably annotate what every parameter means in the callback

Reply:
And how do you do that?

Reply:
oh, i just meant writing it in the documentation

Reply:
Oh ok

Reply:
I also published vach 0.2.3. The changes are insignificant

Reply:
Okay, could you add a commit to this branch that adds some documentation?

Reply:
Leave that to the next commit. This is technically the last commit from my fork before I replace it with a working fork